manager/integration/tests/test_upgrade.py

```diff
@@ -45,39 +62,111 @@ def test_upgrade(upgrade_image_tag, volume_name):  # NOQA
+                                    pod_make, pod_volume_name,
+                                    data_path=pod_data_path,
+                                    add_liveness_prope=False)
+
+    # Create multiple volumes used by StatefulSet
+    statefulset_name = 'statefulset-upgrade-test'
+    update_statefulset_manifests(statefulset,
+                                 storage_class,
+                                 statefulset_name)
+    create_storage_class(storage_class)
+    create_and_wait_statefulset(statefulset)
+    statefulset_pod_info = get_statefulset_pod_info(core_api, statefulset)
+
+    # upgrade Longhorn
+    assert longhorn_upgrade(upgrade_image_tag)
 
-    volume = create_and_check_volume(client,
-                                     volume_name,
-                                     size=SIZE)
+    client = get_longhorn_api_client()
 
-    volume.attach(hostId=host_id)
+    # wait for 1 minute before checking pod restarts
+    time.sleep(60)
 
-    volume = wait_for_volume_healthy(client, volume_name)
+    pod = core_api.read_namespaced_pod(name=pod_name,
+                                       namespace='default')
+    assert pod.status.container_statuses[0].restart_count == 0
 
-    data = write_volume_random_data(volume)
+    for sspod_info in statefulset_pod_info:
+        sspod = core_api.read_namespaced_pod(name=sspod_info['pod_name'],
+                                             namespace='default')
+        assert \
+            sspod.status.container_statuses[0].restart_count == 0
 
-    assert longhorn_upgrade(upgrade_image_tag)
+    statefulset['spec']['replicas'] = replicas = 0
+    apps_api = get_apps_api_client()
 
-    client = get_longhorn_api_client()
+    apps_api.patch_namespaced_stateful_set(
+        name=statefulset_name,
+        namespace='default',
+        body={
+            'spec': {
+                'replicas': replicas
+            }
+        })
 
-    volume = client.by_id_volume(volume_name)
+    delete_and_wait_pod(core_api, pod_name)
 
+    volume = client.by_id_volume(volume_name)
     volume.detach()
 
-    volume = wait_for_volume_detached(client, volume_name)
+    volumes = client.list_volume()
+
+    for v in volumes:
+        wait_for_volume_detached(client, v.name)
 
     engineimages = client.list_engine_image()
 
     for ei in engineimages:
         if ei.image == new_ei_name:
             new_ei = ei
 
-    volume.engineUpgrade(image=new_ei.image)
+    volumes = client.list_volume()
+
+    for v in volumes:
+        volume = client.by_id_volume(v.name)
+        volume.engineUpgrade(image=new_ei.image)
+
+    statefulset['spec']['replicas'] = replicas = 2
+    apps_api = get_apps_api_client()
+
+    apps_api.patch_namespaced_stateful_set(
+        name=statefulset_name,
+        namespace='default',
+        body={
+            'spec': {
+                'replicas': replicas
+            }
+        })
+
+    wait_statefulset(statefulset)
```

Comment:
We need to check the statefulset data. (do we have a way to do it?)

Reply:
yes, fixed